Sources/SwiftIfConfig/StaticBuildConfiguration.swift

```diff
@@ -0,0 +1,402 @@
+  /// the `hasFeature` syntax, e.g.,
+  ///
+  /// ```swift
+  /// #if hasFeature(VariadicGenerics)
+  /// // ...
+  /// #endif
+  /// ```
+  public var features: Set<String> = []
+
+  /// The set of attributes that are available.
+  ///
+  /// Attributes are determined by the Swift compiler. They can be checked
+  /// with `hasAttribute` syntax, e.g.,
+  ///
+  /// ```swift
+  /// #if hasAttribute(available)
+  /// // ...
+  /// #endif
+  /// ```
+  public var attributes: Set<String> = []
+
+  /// The active target OS names, e.g., "Windows", "iOS".
+  public var targetOSNames: Set<String> = []
+
+  /// The active target architectures, e.g., "x64_64".
+  public var targetArchitectures: Set<String> = []
+
+  /// The active target environments, e.g., "simulator".
+  public var targetEnvironments: Set<String> = []
+
+  /// The active target runtimes, e.g., _ObjC.
+  public var targetRuntimes: Set<String> = []
+
+  /// The active target's pointer authentication schemes, e.g., "arm64e".
+  public var targetPointerAuthenticationSchemes: Set<String> = []
+
+  /// The active target's object file formats, e.g., "COFF"
+  public var targetObjectFileFormats: Set<String> = []
+
+  /// The bit width of a data pointer for the target architecture.
+  ///
+  /// The target's pointer bit width (which also corresponds to the number of
+  /// bits in `Int`/`UInt`) can only be queried with the experimental syntax
+  /// `_pointerBitWidth(_<bitwidth>)`, e.g.,
+  ///
+  /// ```swift
+  /// #if _pointerBitWidth(_32)
+  /// // 32-bit system
+  /// #endif
+  /// ```
+  public var targetPointerBitWidth: Int = 64
+
+  /// The atomic bit widths that are natively supported by the target
+  /// architecture.
+  ///
+  /// This lists all of the bit widths for which the target provides support
+  /// for atomic operations. It can be queried with
+  /// `_hasAtomicBitWidth(_<bitwidth>)`, e.g.
+  ///
+  /// ```swift
+  /// #if _hasAtomicBitWidth(_64)
+  /// // 64-bit atomics are available
+  /// #endif
+  public var targetAtomicBitWidths: [Int] = []
+
+  /// The endianness of the target architecture.
+  ///
+  /// The target's endianness can onyl be queried with the experimental syntax
+  /// `_endian(<name>)`, where `<name>` can be either "big" or "little", e.g.,
+  ///
+  /// ```swift
+  /// #if _endian(little)
+  /// // Swap some bytes around for network byte order
+  /// #endif
+  /// ```
+  public var endianness: Endianness = .little
+
+  /// The effective language version, which can be set by the user (e.g., 5.0).
+  ///
+  /// The language version can be queried with the `swift` directive that checks
+  /// how the supported language version compares, as described by
+  /// [SE-0212](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0212-compiler-version-directive.md). For example:
+  ///
+  /// ```swift
+  /// #if swift(>=5.5)
+  /// // Hooray, we can use tasks!
+  /// ```
+  public var languageVersion: VersionTuple
```

Comment:
Shouldn’t this be called `languageMode` to align with https://github.com/swiftlang/swift-evolution/blob/main/proposals/0441-formalize-language-mode-terminology.md?

Reply:
Ah, forgot about that again. Maybe we need to revise the terminology in `BuildConfiguration` at some point…